presto-main/src/main/java/com/facebook/presto/server/WorkerResource.java

```diff
@@ -78,6 +78,33 @@ public Response getStatus(@PathParam("nodeId") String nodeId)
         return Response.ok(responseStream, APPLICATION_JSON_TYPE).build();
     }
 
+    @GET
+    @Path("{nodeId}/thread")
+    public Response getThreads(@PathParam("nodeId") String nodeId)
+    {
+        Set<Node> nodes = nodeManager.getNodes(NodeState.ACTIVE);
+        Node node = nodes.stream()
+                .filter(n -> n.getNodeIdentifier().equals(nodeId))
+                .findFirst()
+                .orElse(null);
+
+        if (node == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        Request request = prepareGet()
+                .setUri(uriBuilder()
+                        .scheme("http")
+                        .hostAndPort(fromParts(node.getHostAndPort().getHostText(), node.getHostAndPort().getPort()))
+                        .appendPath("v1")
+                        .appendPath("thread")
+                        .build())
+                .build();
+
+        InputStream responseStream = httpClient.execute(request, new StreamingJsonResponseHandler());
```

Comment:
This whole method looks the same as the other one, except for the path. You could probably dedupe the whole thing